src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1042,9 +1064,95 @@ class DistributeByClauseSegment(BaseSegment):
     )
 
 
+@spark3_dialect.segment()
+class HintFunctionSegment(BaseSegment):
+    """A Function within a SparkSQL Hint.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-hints.html
+    """
+
+    type = "hint_function"
+    match_grammar = Sequence(
+        Sequence(
+            Ref("FunctionNameSegment"),
+            Bracketed(
+                Ref(
+                    "FunctionContentsGrammar",
+                    # Brackets may be empty
+                    optional=True,
+                    ephemeral_name="FunctionContentsGrammar",
+                ),
```

Comment:
You problem is it is finding the table reference and assuming it is a column reference.

If you don't need the full `FunctionContentsGrammar` and this hint will always be just a single table identifier or a number, then you could just change that to this:

```suggestion
                OneOf(Ref("SingleIdentifierGrammar"), Ref("NumericLiteralSegment")),
```

Reply:
Great! Thanks for the suggestion